subprojects/docs/src/docs/userguide/javaPlugin.adoc

```diff
@@ -481,37 +481,42 @@ By default, the Java compiler runs in the Gradle process. Setting `options.fork`
 
 Starting with Gradle 2.1, it is possible to compile Java incrementally. See the api:org.gradle.api.tasks.compile.JavaCompile[] task for information on how to enable it.
 
-Main goals for incremental compilations are:
+The main goals for incremental compilation are:
 
-* Avoid wasting time compiling source classes that don't have to be compiled. This means faster builds, especially when a change to a source class or a jar does not incur recompilation of many source classes that depend on the changed input.
+* Make incremental builds much faster.
 * Change as few output classes as possible. Classes that don't need to be recompiled remain unchanged in the output directory. An example scenario when this is really useful is using JRebel - the fewer output classes are changed the quicker the JVM can use refreshed classes.
 
-The incremental compilation at a high level:
+To help you understand how incremental compilation works, the following provides a high-level overview:
 
-* The stale class detection favors reliability over speed. The algorithm uses bytecode analysis and deals gracefully with compiler optimizations (inlining of non-private constants), transitive class dependencies, etc. Example: When a class with a public constant changes, we eagerly compile classes that use the same constants to avoid problems with constants inlined by the compiler.
-* To make incremental compilation fast, we cache class analysis results and jar snapshots. The initial incremental compilation can be slower due to the cold caches.
+* Gradle will recompile all classes _affected_ by a change.
+* A class is _affected_ if it has been changed or if it depends on another affected class.
```

Comment:
Perhaps:

    A class is _affected_ if its source has changed or if it depends on another affected class, whether that class is in the same project, another project, or in a JAR dependency

You could then remove the item below about incrementally analyzing JAR contents if you want.